scripts/setup-k8s-master.sh.in

```diff
@@ -55,41 +55,61 @@ kubeadm reset --force
 
 cat >/tmp/kubeadm.yaml <<EOF
 ---
-apiVersion: kubeadm.k8s.io/v1alpha2
-kind: MasterConfiguration
-kubernetesVersion: "$(cat /etc/kubernetes_community_ami_version)"
-api:
-  controlPlaneEndpoint: "${LB_DNS}:443"
-apiServerCertSANs:
-- "${LB_IPV4}"
-apiServerExtraArgs:
-  cloud-provider: "aws"
-nodeRegistration:
-  name: "${HOSTNAME}"
-  kubeletExtraArgs:
-    cloud-provider: "aws"
+apiVersion: kubeadm.k8s.io/v1beta1
 bootstrapTokens:
 - groups:
-  - "system:bootstrappers:kubeadm:default-node-token"
+  - system:bootstrappers:kubeadm:default-node-token
   token: "{{ClusterToken}}"
-  ttl: "0s"
+  ttl: 0s
   usages:
   - signing
   - authentication
-controllerManagerExtraArgs:
-  cloud-provider: "aws"
-  allocate-node-cidrs: "false"
+kind: InitConfiguration
+nodeRegistration:
+  criSocket: /var/run/dockershim.sock
+  name: "${HOSTNAME}"
+  kubeletExtraArgs:
+    cloud-provider: "aws"
+  taints:
+  - effect: NoSchedule
+    key: node-role.kubernetes.io/master
+---
+apiServer:
+  timeoutForControlPlane: 4m0s
+  certSANs:
+    - "${LB_IPV4}"
+  extraArgs:
+    cloud-provider: "aws"
+apiVersion: kubeadm.k8s.io/v1beta1
+certificatesDir: /etc/kubernetes/pki
```

Comment:
some of these look like defaults, is there a reason the defaults are specified?

Reply:
For the most part I just went line by line through the old config and found the new versions.

I think a few defaults may have gotten though because I was also referencing the default config that kubeadm suggested. I can try and prune those a bit more if you think I should.

Reply:
Removed the defaults I could find.